acceptance_testing.go

```diff
@@ -1002,8 +1026,39 @@ func (a acceptanceTest) isEqualRecords(is *is.I, want, got []Record) {
 }
 
 func (a acceptanceTest) isEqualRecord(is *is.I, want, got Record) {
-	// TODO figure out if we should compare anything else or at least enforce
-	//  some properties for other fields (e.g. not nil)
-	is.Equal(want.Key, got.Key)         // record key did not match (want != got)
-	is.Equal(want.Payload, got.Payload) // record payload did not match (want != got)
+	a.isEqualData(is, want.Key, got.Key)
+	a.isEqualData(is, want.Payload, got.Payload)
+
+	// check metadata fields, if they are there they should match
+	if len(got.Metadata) == 0 {
+		is.Equal(got.Metadata, nil) // if there is no metadata the map should be nil
+	} else {
+		for k, wantMetadata := range want.Metadata {
+			if gotMetadata, ok := got.Metadata[k]; ok {
+				// only compare fields if they actually exist
+				is.Equal(wantMetadata, gotMetadata) // record metadata did not match (want != got)
+			}
+		}
+	}
+
+	is.True(!want.CreatedAt.After(got.CreatedAt)) // the expected CreatedAt timestamp shouldn't be after the actual CreatedAt timestamp
```

Comment:
nit: Would this read better if it was `want.CreatedAt.Before(got.CreatedAt)` instead of the negated `After`?